src/bindings/python/nixl-meta/nixl_ep/__init__.py

```
@@ -0,0 +1,77 @@
+                raise
+            raise ImportError(
+                f"torch reports CUDA {cuda_major} but {pip_name} is not installed"
+            ) from e
+    # CPU-only torch — use whatever backend is installed
+    for mod_name in ("nixl_ep_cu13", "nixl_ep_cu12"):
+        try:
+            return importlib.import_module(mod_name).__name__
+        except ModuleNotFoundError as e:
+            if e.name != mod_name:
+                raise
+            continue
```

Ruff (0.15.12)

[warning] `try`-`except` within a loop incurs performance overhead

(PERF203)